### PR TITLE
Watch for elm.json changes

### PR DESCRIPTION
**Pull request by jmbockhorst**
Not sure if the progress is working. It wasn't showing up in my test but it might have finished too fast.

### CODE REVIEW

File: src/elmWorkspace.ts [hunk 1]
```diff
@@ -385,6 +388,24 @@ export class ElmWorkspace implements IElmWorkspace {
     const pathToElmJson = path.join(this.rootPath.fsPath, "elm.json");
     this.connection.console.info(`Reading elm.json from ${pathToElmJson}`);
 
+    await this.elmJsonWatcher?.close();
+    this.elmJsonWatcher = chokidar.watch(pathToElmJson).on("change", () => {
```
Comment (Razzeee) [inline src/elmWorkspace.ts:392]: We might want to debounce here

**Comment by Razzeee** [comment]
No, I don't think #32 is closed, we still miss if the elm package cache changes.

- Checkout an elm project from git
- Open it in vscode
- The server starts
- You run `elm make` for the first time and get dependencies
- The server doesn't know about them

**Comment by jmbockhorst** [comment]
Should we run `elm make` on init to get dependencies? The server won't even start otherwise.

**Comment by Razzeee** [comment]
Might be a way.

I feel like the old behavior would be better, if the server can start everything and we start to watch the package folder for changes. Not a huge fan of watching a folder in such a central location, but what can we do.

File: src/elmWorkspace.ts (hunk 1 above)
Comment (jmbockhorst) [reply src/elmWorkspace.ts:392]: I'm not sure how much the file would be saved many times quickly. Since is just disk changes, not editor changes.

**Comment by jmbockhorst** [comment]
What do you mean by old behavior?

**Comment by Razzeee** [comment]
The old behavior indexed what was there (but wrongish) but didn't fail if a package was missing. It ended up without that package in it's indexes, which causes inconsistencies, but it was still useable. Describing how it was before the new package resolve algorithm.

**Comment by jmbockhorst** [comment]
Alright, so maybe the dependency solver algorithm should look at the packages on Github (if online) or wherever the compiler gets them. That way it can validate the package versions like the compiler would, and get the correct version no matter what is on disk. Then we can just ignore if the package is missing on disk, like before. It would still fail if the dependency versions were messed up, just like the compiler would fail. But I think that is fine because once they fix it, we will reload the server.

**Comment by jmbockhorst** [comment]
Fixed the progress reporting and tested it. This is working nicely now.

File: src/elmWorkspace.ts (hunk 1 above)
Comment (Razzeee) [reply src/elmWorkspace.ts:391]: Didn't know that, then again, editors like vim seem to change with every keystroke?

**Comment by Razzeee** [comment]
Test suite seems to run weirdly long

File: src/elmWorkspace.ts (hunk 1 above)
Comment (jmbockhorst) [reply src/elmWorkspace.ts:391]: We should investigate, but I wouldn't think they would save to disk every keystroke.

Comment (Razzeee) [reply src/elmWorkspace.ts]: I know that a user reported that, but it might be down to the vim config

**Pull request merged**